components-core/src/main/java/com/adyen/checkout/components/repository/OrderStatusRepository.kt

```diff
@@ -32,11 +31,10 @@ class OrderStatusRepository {
         Logger.d(TAG, "Getting order status")
         try {
             val request = OrderStatusRequest(orderData)
-            return OrderStatusConnection(
+            return OrderStatusService(configuration.environment).getOrderStatus(
                 request,
-                environment = configuration.environment,
-                clientKey = configuration.clientKey
-            ).suspendedCall()
+                configuration.clientKey
+            )
         } catch (e: IOException) {
```

Comment:
Instead of try catch and throwing a `CheckoutException`, should we return a wrapper (`sealed class ApiResult` or something)? Also keep in mind that we might need to parse the response body for some calls and extract values from it

Reply:
Like the idea! It's a bit of a behavioural change though, so I was thinking maybe we can merge this PR already into develop and do that on a separate branch?

Reply:
👍

Reply:
Was also thinking about `Result`, but a sealed class for errors wouldn't really work with it 🤔